drivers/encx24j600/encx24j600.c

```diff
@@ -82,12 +82,19 @@ static inline void unlock(encx24j600_t *dev) {
 }
 
 void encx24j600_setup(encx24j600_t *dev, const encx24j600_params_t *params)
```

Comment:
Why is this separate function needed?

Reply:
I'm deprecating this

Reply:
I don't understand - why not just add an index parameter to `encx24j600_setup()` like it was done with the other drivers?

Reply:
That's an API change, I did not want to break it

Reply:
That's silly - the API is only used internally and it's a really simple change.
We did not go though a deprecation period to add the parameter to the other drivers.

Reply:
actually I'm using that API externally :/ I did not notice the change before because I'm only using encx24j600 and netdev_tap.

Reply:
> That's silly - the API is only used internally and it's a really simple change. We did not go though a deprecation period to add the parameter to the other drivers.

The fact that it was not done for other drivers does not make it correct. This is publicly exposed in the header file, so not reserved for internal usage only. I'm not a fan of such API breaks, as out-of-tree modules (or network stacks) may be relaying on this.

Reply:
I guess they could just

```C
#if RIOT_VERSION_CODE < RIOT_VERSION_NUM(2022, 4, 0, 0)
    encx24j600_setup(dev, params);
#else  
    encx24j600_setupt(dev, params, 0);
#endif
```

but if you prefer to do it that way I won't object.

Reply:
@kaspar030 as an external user of this, what do you find to be the better migration path?

Reply:
well, apparrently the registering of netdevs has been in master for years, I just didn't realize because encx24j600 wasn't updated.

To not break existing code, I would have prefer to keep `encx24j600_setup()` as-is and add a new function doing _setup() and the netdev_register().

Reply:
> To not break existing code, I would have prefer to keep `encx24j600_setup()` as-is and add a new function doing _setup() and the netdev_register().

But mind that then you'll have an inconsistency in function naming in regards to the other drivers.

Reply:
Yes, consistency is king definitely.

I have only one user, I guess I can just no-op the `netdev_register()`.

Reply:
Ok, changed in the last commit. I will be doing the same for w5100 then

Reply:
Done